bindings/cs/MongoDB.Crypt/AwsKeyId.cs

```diff
@@ -33,8 +35,24 @@ public AwsKeyId(string customerMasterKey, string region)
         {
             Region = region;
             CustomerMasterKey = customerMasterKey;
+            AlternateKeyNames = new List<byte[]>().AsReadOnly();
         }
 
+        /// <summary>
+        /// Creates an <see cref="AwsKeyId"/> class.
+        /// </summary>
+        /// <param name="customerMasterKey">The customerMasterKey.</param>
+        /// <param name="region">The region.</param>
+        /// <param name="alternateKeyNames">The alternate key names.</param>
+        public AwsKeyId(string customerMasterKey, string region, IEnumerable<byte[]> alternateKeyNames)
+        {
+            Region = region;
+            CustomerMasterKey = customerMasterKey;
+            AlternateKeyNames = alternateKeyNames;
+        }
+
+        public IEnumerable<byte[]> AlternateKeyNames { get; }
```

Comment:
Some properties are defined before the constructor and some after.

Never mind the fact that automatic properties are evil...

Reply:
Why are key names of type `byte[]` and not `string`?

Reply:
Do we want the type to be `IEnumerable<T>` or `IReadOnlyList<T>`?

Reply:
Sorry, I initially was trying to minimize the formatting changes made to preserve the history of the file! I've updated the declaration order as well as the XML docstrings regarding key alternate names (Each byte array describes an alternative key name via a BsonDocument in the following format: { "keyAltName" : [BSON UTF8 value] }).

Reply:
Type updated per your suggestion!

Reply:
In the interest of saving time, I'd like to leave the auto properties as is rather than refactor all classes to use backing fields if you have no objections.